applications/crossbar/src/modules/cb_configs.erl

```diff
@@ -15,198 +12,112 @@
+
+validate(Context, ?HTTP_PATCH, ConfigId) ->
+    Parent = kapps_config_util:get_config(cb_context:account_id(Context), ConfigId),
+    validate_with_parent(Context, ConfigId, Parent);
+
+validate(Context, ?HTTP_POST, ConfigId) ->
+    Parent = kapps_config_util:get_reseller_config(cb_context:account_id(Context), ConfigId),
+    validate_with_parent(Context, ConfigId, Parent).
+
+-spec validate_with_parent(cb_context:context(), ne_binary(), kz_json:object()) -> cb_context:context().
+validate_with_parent(Context, ConfigId, Parent) ->
+    RequestData = strip_id(cb_context:req_data(Context)),
+    FullConfig = kz_json:merge_recursive(Parent, RequestData),
+    Schema = kapps_config_util:account_schema_name(ConfigId),
+    cb_context:validate_request_data(Schema, cb_context:set_req_data(Context, FullConfig),
+                                     fun(Ctx) ->
+                                             cb_context:set_req_data(Ctx, kz_json:diff(RequestData, Parent))
+                                     end
+                                    ).
+
 -spec put(cb_context:context(), path_token()) -> cb_context:context().
-put(Context, _) ->
-    crossbar_doc:save(Context).
-
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%% If the HTTP verib is POST, execute the actual action, usually a db save
-%% (after a merge perhaps).
-%% @end
-%%--------------------------------------------------------------------
--spec post(cb_context:context(), path_token()) -> cb_context:context().
-post(Context, _) ->
-    crossbar_doc:save(Context).
-
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%% If the HTTP verib is PATCH, execute the actual action, usually a db save
-%% (after a merge).
-%% @end
-%%--------------------------------------------------------------------
+put(Context, ConfigId) -> post(Context, ConfigId).
+
 -spec patch(cb_context:context(), path_token()) -> cb_context:context().
-patch(Context, _) ->
-    crossbar_doc:save(Context).
-
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%% If the HTTP verib is DELETE, execute the actual action, usually a db delete
-%% @end
-%%--------------------------------------------------------------------
+patch(Context, ConfigId) -> post(Context, ConfigId).
+
+-spec post(cb_context:context(), path_token()) -> cb_context:context().
+post(Context, ConfigId) ->
+    Stored = kz_json:private_fields(kapps_account_config:get(cb_context:account_id(Context), ConfigId)),
+    case {cb_context:req_data(Context), kz_doc:revision(Stored)} of
+        {?EMPTY_JSON_OBJECT, undefined} -> skip;
+        {?EMPTY_JSON_OBJECT, _} ->
+            crossbar_doc:delete(cb_context:set_doc(Context, Stored));
+        {Diff, _} ->
+            crossbar_doc:save(Context, kz_json:merge_recursive(Stored, Diff), [])
+    end,
+    set_config_to_context(ConfigId, Context).
+
```

Comment:
can we move this common handling to a new fun (save, save_config, or other name) and call that fun from here, from put and from patch.